set AuthScheme.OAuth to `Bearer` as defined in RFC6750

hardysim opened the pull request:
In RFC6750 the auth-scheme is defined as `Bearer` (not `OAuth`). See https://tools.ietf.org/html/rfc6750#section-3:

> All challenges defined by this specification MUST use the auth-scheme value "Bearer".

This is also commonly used by servers which respond with `Bearer` as well so this must match to be used by an (not yet implemented) `OAuthAuthProvider` (see https://github.com/ktorio/ktor/issues/1595).

hardysim commented:
Any progress on this? It's not that much to review ;)

cy6erGn0m commented:
Sorry for the huge delay. But I see the problem: this value is passed in OAuth1.a implementation when obtaining a token. So it couldn't be simply changed. Most likely the RFC number is wrong that makes confusion. I believe we simply need to add Bearer constant instead of replacing the existing one

hardysim commented:
@cy6erGn0m which RFC do you mean? I changed RFC-6749 to RFC-6750 but both are about OAuth 2.0. I want to use this on OAuth 2.0 (only - I don't think anyone is still using OAuth 1).

Do you mean this constant is used in multiple places (and one of it is the OAuth 1 flow)? Then a second constant would be better, correct.

---

I see that #1740 has been merged already. In there, I added an extension-property `AuthScheme.Bearer` so there we have our additional constant. Just not in a "nice" place.

But I think it has been merged into `e5l/develop` only so it might not be in the actual `master`.

@e5l Any updates on this? [You mentioned, you wanted to check binary compatibility and add some tests](https://github.com/ktorio/ktor/pull/1740#issuecomment-604839760).

Should we add `AuthScheme.Bearer` directly to the `AuthScheme.kt`-file? If so, this (#1733) MR would nicely complement my code from #1740 and we could merge both to `master` if you're happy with it.

LouisCAD commented:
Why has this issue been closed, @e5l?

e5l commented:
I reopened the related YT issue, will fix it in the next minor.

hardysim commented:
> I reopened the related YT issue, will fix it in the next minor.

Would you please link the YT issue so we can keep track of the progress? Thanks!

LouisCAD commented:
Yep, please, always link the related issue.
